commands/equip_commands.py

```diff
@@ -0,0 +1,258 @@
+
+    def func(self):
+        "implements the command."
+
+        caller = self.caller
+        args = self.args.strip()
+
+        if not args:
+            caller.msg("Wear what?")
+            return
+
+        # this will search for a target
+        obj = caller.search(args)
+
+        if not obj:
+            return
+
+        if not obj in caller.contents:
+            caller.msg("You don't have %s in your inventory.")
+
+        # equip primary hand with the proper command
+        if 'right hand' in utils.make_iter(obj.db.slot):
+            caller.execute_cmd("wield %s" % args)
+            return
+
+        if not obj.db.slot or 'holds' in utils.make_iter(obj.db.slot)  \
+           or not obj.access(caller, 'equip'):
+            caller.msg("You can't equip %s." % obj)
+            return
+
+        if obj in caller.equip.items():
+            caller.msg("You already equip %s." % obj.name)
+            return
+
+        if not caller.equip.add(obj):
+            string = "You can't equip %s." % obj
+            if [s for s in utils.make_iter(obj.db.slot) if s in caller.equip.empty_slots()]:
+                string += " You have already something there."
+            caller.msg(string)
+            return
+        # call hook
+        if hasattr(obj, "at_equip"):
+            obj.at_equip(caller)
+        caller.msg("You wear %s." % obj)
+        caller.location.msg_contents("%s wears %s." % (caller.name.capitalize(), obj),
+                                     exclude=caller)
+
+
+class CmdWield(Command):
+    """
+    wield
+
+    Usage:
+      wield <obj>
+
+    Wield an object.
+
+    """
+    key = "wield"
+    aliases = ["equip primary", "ep"]
+    locks = "cmd:all()"
+
+    def func(self):
+        "implements the command."
+        caller = self.caller
+        args = self.args.strip()
+
+        if not args:
+            caller.msg("Wield what?")
+            return
+
+        # this will search for a target
+        obj = caller.search(args)
+
+        if not obj:
+            return
+
+        if not obj in caller.contents:
+            caller.msg("You don't have %s in your inventory.")
+            return
+
+        if not obj.db.slot                                         \
+           or 'right hand' not in utils.make_iter(obj.db.slot)     \
+           or not obj.access(caller, 'equip'):
+            caller.msg("You can't wield %s." % obj)
+            return
+
+        if obj in caller.equip.items():
+            caller.msg("You already wield %s." % obj.name)
+            return
+
+        if not caller.equip.add(obj):
+            caller.msg("You can't wield %s." % obj)
+            return
+
+        # call hook
+        if hasattr(obj, "at_equip"):
+            obj.at_equip(caller)
+
+        caller.msg("You wield %s." % obj)
+        caller.location.msg_contents("%s wields %s." % (caller.name.capitalize(), obj),
+                                     exclude=caller)
+
+class CmdHold(Command):
```

Comment:
Similarly with Hold, it shares much with Wear and Wield. Can we refactor so Wear, Wield, and Hold use functionality from a more abstract helper class/function(s)?